### PR TITLE
Adding debian build

### PR DESCRIPTION
**Pull request by allenyin96**
1. Added buildDeb config in build.gradle
2. Created a new file, packaging/deb/postinst for debian post install

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -212,13 +210,23 @@ afterEvaluate {
...
 
     buildRpm {
+        arch = 'NOARCH'
+        postInstall file('packaging/rpm/postinst')
         archiveName "${packageName}-${version}.rpm"
         dependsOn 'assemble'
     }
+
+    buildDeb {
+        arch = 'amd64'
```
Comment (vinooamzn) [inline build.gradle]: rpm is NOARCH but deb is amd64? Can we keep them consistent?

File: packaging/deb/postinst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  IS_UPGRADE=true
+fi
+
+#DEBHELPER#
+
+if [ "x$IS_UPGRADE" != "xtrue" ]; then
+    if command -v systemctl > /dev/null; then
+        echo '# Enabling opendistro performance analyzer to start and stop along with elasticsearch.service'
+        systemctl daemon-reload
+        systemctl enable opendistro-performance-analyzer.service || true
```
Comment (vinooamzn) [inline packaging/deb/postinst]: Should we enable the service on upgrade if it was disabled in the previous version?

Comment (allenyin96) [reply packaging/deb/postinst]: Correct me if I'm wrong, but seems to me this block (line 26 to line 36) is only handling the case of a fresh install. If we are handling an upgrade, my opinion is that we should leave the previous settings as they are, which is what we are doing in this script.

File: build.gradle (hunk 1 above)
Comment (allenyin96) [reply build.gradle]: In the package we're using (ospackage), there is no NOARCH option for deb build's arch. See the arch entry in this page: https://github.com/nebula-plugins/gradle-ospackage-plugin/wiki/Deb-Plugin

**Pull request merged**